mockgcp/mockcompute/instancesv1.go

```diff
@@ -0,0 +1,356 @@
+	// }
+
+	if err := s.storage.Create(ctx, fqn, obj); err != nil {
+		return nil, status.Errorf(codes.Internal, "error creating instance: %v", err)
+	}
+
+	return s.newLRO(ctx, name.Project.ID)
+}
+
+// Updates a Instance resource in the specified project using the data included in the request.
+func (s *InstancesV1) Update(ctx context.Context, req *pb.UpdateInstanceRequest) (*pb.Operation, error) {
+	reqName := "projects/" + req.GetProject() + "/zones/" + req.GetZone() + "/instances/" + req.GetInstance()
+	name, err := s.parseZonalInstanceName(reqName)
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+	obj := &pb.Instance{}
+	if err := s.storage.Get(ctx, fqn, obj); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "instance %q not found", fqn)
+		}
+		return nil, status.Errorf(codes.Internal, "error reading instance: %v", err)
+	}
+
+	// TODO: Implement helper to implement the full rules here
+	proto.Merge(obj, req.GetInstanceResource())
+
+	if err := s.storage.Update(ctx, fqn, obj); err != nil {
+		return nil, status.Errorf(codes.Internal, "error updating instance: %v", err)
+	}
+
+	return s.newLRO(ctx, name.Project.ID)
+}
+
+func (s *InstancesV1) AttachDisk(ctx context.Context, req *pb.AttachDiskInstanceRequest) (*pb.Operation, error) {
+	reqName := "projects/" + req.GetProject() + "/zones/" + req.GetZone() + "/instances/" + req.GetInstance()
+	name, err := s.parseZonalInstanceName(reqName)
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+	obj := &pb.Instance{}
+	if err := s.storage.Get(ctx, fqn, obj); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "instance %q not found", fqn)
+		}
+		return nil, status.Errorf(codes.Internal, "error reading instance: %v", err)
+	}
+
+	attachedDisk := proto.Clone(req.GetAttachedDiskResource()).(*pb.AttachedDisk)
+	attachedDisk.Kind = PtrTo("compute#attachedDisk")
+	attachedDisk.Index = PtrTo(int32(len(obj.Disks)))
+
+	obj.Disks = append(obj.Disks, attachedDisk)
+
+	if err := s.storage.Update(ctx, fqn, obj); err != nil {
+		return nil, status.Errorf(codes.Internal, "error updating instance: %v", err)
+	}
+
+	return s.newLRO(ctx, name.Project.ID)
+}
+
+func (s *InstancesV1) DetachDisk(ctx context.Context, req *pb.DetachDiskInstanceRequest) (*pb.Operation, error) {
+	reqName := "projects/" + req.GetProject() + "/zones/" + req.GetZone() + "/instances/" + req.GetInstance()
+	name, err := s.parseZonalInstanceName(reqName)
+	if err != nil {
+		return nil, err
+	}
+
+	fqn := name.String()
+	obj := &pb.Instance{}
+	if err := s.storage.Get(ctx, fqn, obj); err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil, status.Errorf(codes.NotFound, "instance %q not found", fqn)
+		}
+		return nil, status.Errorf(codes.Internal, "error reading instance: %v", err)
```

Comment:
```suggestion
		return nil, err
```

Reply:
Probably going to be easier for you to just make the changes locally, rather than applying the changes here and squashing, so I think you get the idea :-)